Sources/Sentry/Profiling/SentryLegacyProfiler.mm

```diff
@@ -0,0 +1,92 @@
+#import "SentryLegacyProfiler.h"
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+
+#    import "SentryLog.h"
+#    import "SentryMetricProfiler.h"
+#    import "SentryProfiledTracerConcurrency.h"
+#    import "SentryProfiler+Private.h"
+#    include <mutex>
+
+#    pragma mark - Private
+
+namespace {
+/** @warning: Must be used from a synchronized context. */
+std::mutex _unsafe_gLegacyProfilerLock;
+
+/** @warning: Must be used from a synchronized context. */
+SentryProfiler *_Nullable _unsafe_gLegacyProfiler;
+
+/** @warning: Must be called from a synchronized context. */
+BOOL
+_unsafe_isRunning(void)
+{
+    return _unsafe_gLegacyProfiler != nil && [_unsafe_gLegacyProfiler isRunning];
+}
+} // namespace
+
+@implementation SentryLegacyProfiler
+
+#    pragma mark - Public
+
++ (BOOL)startWithTracer:(SentryId *)traceId
+{
+    std::lock_guard<std::mutex> l(_unsafe_gLegacyProfilerLock);
+
+    if (_unsafe_isRunning()) {
+        SENTRY_LOG_DEBUG(@"A legacy profiler is already running.");
+        sentry_trackProfilerForTracer(_unsafe_gLegacyProfiler, traceId);
+        // record a new metric sample for every concurrent span start
+        [_unsafe_gLegacyProfiler.metricProfiler recordMetrics];
+        return YES;
+    }
+
+    _unsafe_gLegacyProfiler = [[SentryProfiler alloc] initWithMode:SentryProfilerModeLegacy];
+    if (_unsafe_gLegacyProfiler == nil) {
+        SENTRY_LOG_WARN(@"Legacy profiler was unable to be initialized, will not proceed.");
+        return NO;
+    }
+
+    sentry_trackProfilerForTracer(_unsafe_gLegacyProfiler, traceId);
+    return YES;
+}
+
++ (BOOL)isCurrentlyProfiling
+{
+    std::lock_guard<std::mutex> l(_unsafe_gLegacyProfilerLock);
+    return [_unsafe_gLegacyProfiler isRunning];
```

Comment:
`l`: reusing the existing method:
```suggestion
    return _unsafe_isRunning();
```

Reply:
Thanks for catching this. Honestly we don't even need the nil check on `_unsafe_gLegacyProfiler` since it's safe to send a message to `nil` and it would return false in that case, so i might simplify this flow.